dace/transformation/dataflow/vectorization.py

```diff
@@ -18,100 +143,341 @@ class Vectorization(transformation.SingleStateTransformation):
-                        if param in symbols:
-                            if array.strides[idx] == 1:
-                                found = True
-                            else:
-                                return False
-            except TypeError:  # cannot determine truth value of Relational
-                return False
+                except TypeError:  # cannot determine truth value of Relational
+                    return False
+
+            for e in state.in_edges(node):
+                # Check for valid copies from other tasklets and/or streams
+                if e.data.data is not None:
+                    src_node = state.memlet_path(e)[0].src
+                    if not isinstance(src_node, (nodes.Tasklet, nodes.AccessNode)):
+                        # Make sure we only have Code->Code copies and from arrays
+                        return False
+
+                    if isinstance(src_node, nodes.AccessNode):
+                        src_desc = src_node.desc(sdfg)
+                        if isinstance(src_desc, data.Stream) and self.target == dtypes.ScheduleType.SVE_Map:
+                            # Stream pops are not implemented
+                            return False
+
+        # Check if it is possible to vectorize data container
+        if self.target == dtypes.ScheduleType.FPGA_Device and self._level == 0:
+            maps_to_vectorize, data_descriptors_to_vectorize = collect_maps_to_vectorize(sdfg, state, map_entry)
+
+            old_map_entry = self.map_entry
+            self._level = 1  # To prevent infinte loop
+
+            for m in maps_to_vectorize:
+                self.map_entry = m
+
+                if not self.can_be_applied(get_innermost_state_for_node(sdfg, m), expr_index,
+                                           get_innermost_sdfg_for_node(sdfg, m), permissive):
+                    return False
+
+            # Check alls strideds of the arrays
+            for a in data_descriptors_to_vectorize:
+
+                correct_sdfg = get_innermost_sdfg_for_array(sdfg, a)
+                array = correct_sdfg.arrays[a]
+                strides_list = list(array.strides)
+
+                if strides_list[-1] != 1:
+                    return False
+
+                strides_list.pop()
+
+                for i in strides_list:
+                    if isInt(i) and i % self.vector_len != 0:
+                        return False
+
+            # Check all maps
+            for m in maps_to_vectorize:
+                real_map: nodes.Map = m.map
+                ranges_list = list(real_map.range)
+
+                if isInt(ranges_list[-1][1]) and (ranges_list[-1][1] + 1) % self.vector_len != 0:
+                    return False
 
-        return found
+                if ranges_list[-1][2] != 1:
+                    return False
+
+            # Check all edges
+            for m in maps_to_vectorize:
+
+                correct_state = get_innermost_state_for_node(sdfg, m)
```

Comment:
Same as above: this is wasteful, just collect this information when you're searching for maps